dbms/src/WindowFunctions/IWindowFunction.cpp

```diff
@@ -133,6 +133,44 @@ struct WindowFunctionRowNumber final : public IWindowFunction
     }
 };
 
+struct WindowFunctionFirstValue final : public IWindowFunction
+{
+public:
+    static constexpr auto name = "first_value";
+
+    explicit WindowFunctionFirstValue(const DataTypes & argument_types_)
+        : IWindowFunction(argument_types_)
+    {
+        assert(argument_types_.size() == 1);
```

Comment:
```suggestion
        RUNTIME_CHECK(argument_types_.size() == 1);
```
much safer

Reply:
> much safer

The `assert` is just used for the validation in ut, so I think `assert` is better.

Reply:
> > much safer
> 
> The `assert` is just used for the validation in ut, so I think `assert` is better.

If tidb pass wrong expr, tiflash will crash.
I prefer not to use assert in any case

Reply:
> > > much safer
> > 
> > 
> > The `assert` is just used for the validation in ut, so I think `assert` is better.
> 
> If tidb pass wrong expr, tiflash will crash. I prefer not to use assert in any case

okk